src/main/java/ee/carlrobert/codegpt/credentials/AbstractCredentialsManager.java

```diff
@@ -46,6 +46,9 @@ protected boolean isCredentialSet(String key) {
   }
 
   protected void setCredential(String key, String credential) {
+    if (credential == null) {
+      return;
+    }
     passwordSafe.setPassword(credentialMapping.get(key), credential);
     credentialCache.put(key, credential);
```

Comment:
I got an error here, since I havent set any Azure api key and the map doesnt allow null values

Reply:
Nice catch!